### DISCUSSION TITLE
Subject combination Query

### DISCUSSION THREAD

**Discussion by sumeetaher** [snigi-gupta/AttendingUniversityAtBuffalo#42] [Q&A] [answered]
Hello,
First of all, this is a very good initiative and effort. Thanks a lot.
All of the incoming batch of Fall 2021 are benefiting from this collaborative effort from seniors.
For my first semester, I have the following subjects in mind:
1) Introduction to Operating Systems by Farshad Ghanei
2)Large Scale Distributed Systems by Bina Ramamurhty / Modern Networking Concepts by HongXin Hu 
3)Introduction to Machine learning by Mingchen Gao
4) Analysis of Algorithms by Roger he

Q1=> Is this combination considered to be a very heavy workload ?
Q2 =>I checked Prof Bina on ratemyprof, she seems to be a good professoo. But personally if anyone has taken this subject, could let me know about the experience?
Q3 => Is OS subject important as I have never had OS subject before during bachelors and I dnt have any plans for taking up research after MS.
Q4 => Should I wait till the next semester to take up Analysis of Algorithms by Shi Li ( is the experience and learning that good with Prof Shi li? ) or should I just take it this semester by Roger he so that it benefits my internship interviews?
Q5 => How much effort is expected in a 3 credit seminar ? Also how is an S grade calculated  in our Cumulative grade?
Q6 = > Considering i never had any network class before, which should I choose from the 2nd subject options?

Thanks for continued support and helping students always.
PS: I have read through the course reviews already posted in the repo. They were immensely insightful.

**Comment by snigi-gupta** [comment, marked as answer]
Hi @SumeetAher,

1. YES, this is a heavy workload. OS and MNC are heavy courses. Choose 1 from this. DS, ML and ALGOS are all medium courses. Choose 2 from this. And then choose 1 Easy course or seminar. [Refer this for more help.](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips)
2. This is the first time Bina is taking the course. Personally, I feel you should take DS by Ethan Blanton.
3. If you don't plan to work in OS and are not passionate about it, then don't take that subject. You won't be able to get decent grades, if you are not interested in the subject. Choose subjects that YOU think will help YOU when YOU apply for jobs. There is a reason I am highlighting YOU. Because YOUR background is different from others, so don't follow what your friends are doing, or what everyone is doing. Don't fall for that trap.
4. Either of the algos are good and bad equally. Shi Li give some practical questions too. Roger's approach is more theoretical but both are hard if you don't self-study. I took roger and he is a good professor. There is a language barrier with Chinese profs due to the accent, but you get used to it. I took roger in first sem itself thinking it will help me in internships, but I wasn't able to self-study much and hence I couldn't utilize it properly because I hampered my learning with an unbalanced workload. So make sure you can focus on each course properly.
5. For this efforts[check this out](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips)
6. Choose the one whose coursework, professor, teaching method excites you (audit the course and you will come to know). There were a lot of courses I took, where I had 0 background beforehand. The key is to focus on the basics, ask the TA/prof for help when in doubt, self-study the course and you will be able to do good in courses that you never touched.

If you have further questions, don't hesitate to ask!
